main/webapp/modules/core/scripts/preferences.js

```diff
@@ -95,6 +96,30 @@ $.i18n().load(dictionary, lang);
 $.i18n().locale = lang;
 //End internationalization
 
+Refine.getPreference = function(key, defaultValue) { 
+  if(!thePreferences.hasOwnProperty(key)) { return defaultValue; }
+
+  return thePreferences[key];
+}
+
+Refine.setPreference = function(key, newValue) { 
+  thePreferences[key] = newValue;
+
+  Refine.wrapCSRF(function(token) {
+    $.ajax({
+      async: false,
+      type: "POST",
+      url: "command/core/set-preference?" + $.param({ name: key }),
+      data: {
+        "value" : JSON.stringify(newValue), 
+        csrf_token: token
+      },
+      success: function(data) { },
+      dataType: "json"
+    });
+  });
+}
```

Comment:
Did we not already have those two functions somewhere else? If so, I would just like to be clear on why you needed to copy them over.

Reply:
> Did we not already have those two functions somewhere else? If so, I would just like to be clear on why you needed to copy them over.

@wetneb: right now, we do a small footrint of functions that are copied between the JS of pages: index.js, projects.js and preferences.js.

This like Refine.POSTcsrf(), etc. I'v written this function for v3.4 2 years ago. For the future, I'm looking at a small file to generalize them. Right now, this was helpful to code this issue. But it's really no big deal to keep it or delete it.

Regards, Antoine

Reply:
Ok but I do not get why you need to add those functions to `preferences.js` whereas the PR does not actually add any usage of those functions in this file - what did I miss?

Reply:
> Ok but I do not get why you need to add those functions to `preferences.js` whereas the PR does not actually add any usage of those functions in this file - what did I miss?

@wetneb : you were not ping on that discussion, but if you read [this](https://github.com/OpenRefine/OpenRefine/pull/4759#issuecomment-1099732745) (PR#4759), it will be clearer. It's a learning tool, and it's also code consolidation, factoring, at the cost of code already written and debugged.

Regards, Antoine

Reply:
If those functions were added to `preferences.js` so that the developer could play with them via their Javascript console during development, I would remove them. Or did I still not get it?

Reply:
> If those functions were added to `preferences.js` so that the developer could play with them via their Javascript console during development, I would remove them. Or did I still not get it?

It's not just for that. I have code to use them IN the actual code, but I didn't want to mess this contributors stuff, just give a tool more. I plan to wait till 2 weeksm before submitting it. I somehow seam to think you already have enough on your plate. Right now, I just want to stay focus on Outreachy and help you. So if I can help anyway, you say.

Regards, Antoine

Reply:
@Babi-B: you can remove my code. And I will continue to review the rest of your work.

Regards, Antoine